crates/red_knot_test/src/parser.rs

```diff
@@ -425,6 +443,7 @@ impl<'s> Parser<'s> {
         &mut self,
         lang: &'s str,
         code: &'s str,
+        keyvals: &[(&'s str, Option<&'s str>)],
```

Comment:
nit: is it worth creating an `Item` struct?

```rs
struct Item<'s> {
    key: &'s str,
    value: Option<&'s str>
}
```

then this could be

```rs
        keyvals: &[Item<'s>],
```

Reply:
Ended up deleting this code.

I'm not a huge fan of anonymous tuples myself, but I tend not to mind them so much when their use is "local." And especially, when we aren't storing them anywhere. But I also don't feel strongly.